### PR TITLE
feat(elasticloadbalancingv2): enforce security group inbound rules prop

### LINKED ISSUE #29516: (aws-elasticloadbalancingv2): NetworkLoadbalancer support `EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic`
### Describe the feature

L1Construct support EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic https://github.com/aws/aws-cdk/pull/28878.
But L2Construct not support this.

### Use Case

Use NetorkLoadbalancer with  VPC Endpoint Service,
This parameter allow  ignore security group when traffic from VPC Endpoint.

### Proposed Solution

NetworkLoadbalancer  pass through EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.133.0

### Environment details (OS name and version, etc.)

windows 11

### PR DESCRIPTION
**Pull request by msambol**
Closes #29516.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts [hunk 1]
```diff
@@ -465,6 +475,13 @@ export interface INetworkLoadBalancer extends ILoadBalancerV2, ec2.IVpcEndpointS
    */
   readonly ipAddressType?: IpAddressType;
 
+  /**
+   * Indicates whether to evaluate inbound security group rules for traffic sent to a Network Load Balancer through AWS PrivateLink
+   *
+   * @default on
+   */
+  readonly enforceSecurityGroupInboundRulesOnPrivateLinkTraffic?: EnforceSecurityGroupInboundRulesOnPrivateLinkTraffic;
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:483]:
```suggestion
   * @default true
   */
  readonly enforceSecurityGroupInboundRulesOnPrivateLinkTraffic?: boolean;
```
What about defining it as a boolean? It seems a bit overkill to define an enum for representing `on`/`off` values

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍 Left a suggestion for changing the variable type

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts (hunk 1 above)
Comment (msambol) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:483]: That would make way too much sense. I'll make the change 👍🏼😂

**Comment by msambol** [comment]
@lpizzinidev Let me know what you think of this. I was going down the getter/setter route but it go convoluted with the transformation from boolean to string. I think this might be cleaner?

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts [hunk 2]
```diff
@@ -37,6 +37,13 @@ export interface NetworkLoadBalancerProps extends BaseLoadBalancerProps {
    * @default IpAddressType.IPV4
    */
   readonly ipAddressType?: IpAddressType;
+
+  /**
+   * Indicates whether to evaluate inbound security group rules for traffic sent to a Network Load Balancer through AWS PrivateLink.
+   *
+   * @default on
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:44]:
```suggestion
   * @default true
```

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/README.md [hunk 3]
```diff
@@ -257,6 +257,20 @@ listener.addTargets('AppFleet', {
 });
 ```
 
+### Enforce security group inbound rules on PrivateLink traffic for a Network Load Balancer
+
+You can indicate whether to evaluate inbound security group rules for traffic 
+sent to a Network Load Balancer through AWS PrivateLink. The default is `on`.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/README.md:263]:
```suggestion
sent to a Network Load Balancer through AWS PrivateLink.
The evaluation is enabled by default. 
```

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts [hunk 4]
```diff
@@ -220,13 +228,25 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
...
     if (props.crossZoneEnabled) { this.setAttribute('load_balancing.cross_zone.enabled', 'true'); }
+    this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic =
+      this.transformEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic(props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic);
+  }
+
+  private transformEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic(value: boolean | undefined): string | undefined {
+    if (value !== undefined) {
+      return value ? 'on' : 'off';
+    }
+    return undefined;
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:249]:
```
  private readonly _enforceSecurityGroupInboundRulesOnPrivateLinkTraffic?: boolean;
  
  ...
  
  public get enforceSecurityGroupInboundRulesOnPrivateLinkTraffic(): string | undefined {
    if (this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic === undefined) return undefined;
    return this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic ? 'on' : 'off';
  }
```
What about a private boolean property and a public getter for the transformation?
It seems like a cleaner solution.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks for the changes 👍
Left some suggestions, feel free to improve it further

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts (hunk 4 above)
Comment (msambol) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:249]: You would need a setter though, correct, to set `_enforceSecurityGroupInboundRulesOnPrivateLinkTraffic`? How would you set that in `super()`?

Comment (msambol) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:249]: yeah i'm in a mental 🥨 with these getters/setters

Comment (lpizzinidev) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts:249]: 🤔 Shouldn't this work with the above getter or am I missing something?
```
  super(scope, id, props, {
      ...
      enforceSecurityGroupInboundRulesOnPrivateLinkTraffic: Lazy.string({
        produce: () => this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic,
      }),
    });

  this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic = props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic
```

Comment (msambol) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts]: That's better, thanks man. I was doing the following 👎🏼 -->

```
this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic = props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic;
```

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Review by godwingrs22 — APPROVED** [review]
Thanks [msambol](https://github.com/msambol) for contributing and [lpizzinidev](https://github.com/lpizzinidev) for the review

**Comment by msambol** [comment]
@godwingrs22 Do you mind re-reviewing this? I had to rebase on `main`.

**Review by godwingrs22 — APPROVED** [review]

**Pull request merged**